public/app/features/variables/query/QueryVariableEditor.tsx

```diff
@@ -151,23 +164,25 @@ export class QueryVariableEditorUnConnected extends PureComponent<Props, State>
 
     if (isLegacyQueryEditor(VariableQueryEditor, datasource)) {
       return (
-        <Field label="Query">
+        <Container>
+          <VariableLegend header={'h4'}>Query</VariableLegend>
           <VariableQueryEditor
             key={datasource.uid}
             datasource={datasource}
             query={query}
             templateSrv={getTemplateSrv()}
             onChange={this.onLegacyQueryChange}
           />
-        </Field>
+        </Container>
```

Comment:
instead of changing the legend component, I would just use the new `<Box>` component here (instead of the new Container), and then use the Text (typography) component for the heading

Reply:
I'll wait for #73637 to be merged and then refactor this accordingly 😊